### ISSUE TITLE
PDF to Zebra printer is blurry and/or poor quality

### ISSUE THREAD

**Issue by lite1979** [qzind/tray#503]
Using 2.1 RC7, Windows 10, Zebra GK420d printer, and compared to Adobe results, QZ-Tray's print results are poor by comparison. 

Printing to  "Microsoft Print to PDF", the resulting PDF is not blurry, so I suspect either the Zebra driver or the font associated with the print job vs. printer capability in this case. 

[Drug-print.pdf](https://github.com/qzind/tray/files/3756942/Drug-print.pdf)

**Comment by tresf** [comment]
Here are the details of my testing:
* The PDF is confirmed to be printing vector (`rasterize: false` is being honored), but the font is curtailing at such a small size. Making it bold will help slightly.  I recommend the label is redesigned so that it prints better.  I feel there's room for improvement in regards to print quality, but it would be an upstream issue for PDFBOX.
* I've tested turning off dithering in the driver settings and it did not seem to help this particular problem.
* I've tested the PDF on the latest release and it has the same results.

In summary what worked best (still not the same quality as Adobe) was setting the following in the ZDesigner driver:
* Darkness: 15
* Speed: 1.5/s

If we decide to reopen this, it will need to be filed upstream.  The tests were done [using PDFBOX on command line](https://pdfbox.apache.org/2.0/commandline.html) so QZ Tray is not a factor.

Closing as `wontfix` for now.  We may have to revisit this.

**Comment by tresf** [comment]
We'll take a deeper dive into this.  Assigning to @bberenz and @lite1979.

@bberenz can you provide Lite with a unit test that does the following:
1. Prints something like a JLabel text, optimized for thermal printing (aliasing off)
2. Prints a PDF text of same size, optimized for thermal printing

Then we'll raise the concerns to PDFBOX.

**Comment by lite1979** [comment]
https://issues.apache.org/jira/browse/PDFBOX-4709

**Comment by lite1979** [comment]
Unfortunately, this issue has been labeled as "won't fix" by Apache, and my bug was labeled as a duplicate of this one:

https://issues.apache.org/jira/browse/PDFBOX-3293

The recommendation is to use another font; the user that raised the issue for the link above lists his fonts as [NOTO CJK](https://www.google.com/get/noto/help/cjk/) and [CNS11643](http://data.gov.tw/node/5961)

**Comment by tresf** [comment]
Per recommendation of PDFBOX, filed upstream bug report: https://github.com/AdoptOpenJDK/openjdk-support/issues/45

**Comment by tresf** [comment]
Update:
* OpenJDK has this patched since JDK16, backported to JDK11 since 11.0.11: https://bugs.openjdk.org/browse/JDK-8256264
* PDFBOX changed their API per https://issues.apache.org/jira/browse/PDFBOX-5093 which allows a workaround.
   * This exposed this bug with Calbri font (Patched in JDK17 and backported to JDK11 since 11.0.13) https://bugs.openjdk.org/browse/JDK-8265761.

**Comment by tresf** [comment]
Initial integration is partially broken... For some reason when added to QZ Tray, the page is cut off half-way.
<img width="25%" alt="image" src="https://user-images.githubusercontent.com/6345473/229369376-cb9f19ab-4fec-4372-9847-042c87e339f2.png">

**Comment by tresf** [comment]
We finally have a build for testing: https://github.com/tresf/tray/releases/download/v2.2.3/qz-tray-2.2.2+PDFFONT1-x86_64.exe.

**Comment by tresf** [comment]
Closed via #1108
